### DISCUSSION TITLE
403 for basic auth using options.username and options.password.

### DISCUSSION THREAD

**Discussion by peter-snr** [sindresorhus/got#1727] [Q&A] [answered]
Hi,

I am trying out this library and would like to know how to execute a request using basic authentication.

My curl request that works fine is:

    curl -H 'Accept: application/json' -u username:password https://company.server.com/somepath

But my request using got fails with a 403.

    const apiResponse =
      await got(testUrl, {
        username: 'username',
        password: 'password',
        https: {
          key: fs.readFileSync('./server.key'),
          certificate: fs.readFileSync('./server.pem')
        }
      });

This is returning a 403 response code.  I also tried to set the auth header but that failed as well.  Really do not want to revert to the request library if I can help  it.

Thanks in advance, any examples would be most appreciated.

**Comment by szmarczak** [comment, marked as answer]
I cannot reproduce:

```js
const got = require('got');

(async () => {
    const result = await got('https://httpbin.org/anything', {
        username: 'hello',
        password: 'world'
    }).json();
    
    const valid = Buffer.from(result.headers.Authorization.split(' ')[1], 'base64').toString() === 'hello:world';
    
    console.log(valid); //=> true
})();
```

**Comment by peter-snr** [comment]
Hi, after further analysis, the url is a web server login and requires cookies.  I will try again, thank you for testing.
